packages/@pollyjs/adapter-puppeteer/src/index.js

```diff
@@ -89,13 +89,21 @@ export default class PuppeteerAdapter extends Adapter {
             // Do not intercept preflight requests
             request.continue();
           } else {
-            this.handleRequest({
+            const pollyRequest = await this.handleRequest({
               headers,
               url,
               method,
               body: request.postData(),
               requestArguments: [request]
             });
+
+            const { response } = pollyRequest;
+
+            request.respond({
```

Comment:
This should be moved to `respondToRequest` for consistency. This adapter also no-ops the `onRequestFinished` hook so there might be some more refactoring that needs to be done.

Reply:
Unfortunately it is not possible to move it to `respondToRequest()` because that method is never called. The reason is the empty `onRequestFinished()` override. I could move this call to `onRequestFinished()`.

I agree that there are still some opportunities to improve the code. I would prefer to postpone the work to a follow-up PR.

Reply:
Yeah, I need to think more about the no-op there. A separate PR makes sense 😄